File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/visual/CustomModel.java
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge by LiquidBounce.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.features.module.modules.visual;

import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.ModuleCategory;
import net.ccbluex.liquidbounce.features.module.ModuleInfo;
import me.zywl.fdpclient.value.impl.BoolValue;
import me.zywl.fdpclient.value.impl.IntegerValue;
import me.zywl.fdpclient.value.impl.ListValue;

@ModuleInfo(name = "CustomModel", description = "Custom player.", category = ModuleCategory.VISUAL)
public class CustomModel extends Module {

    public static BoolValue customModel = new BoolValue("CustomModel", false);
    public static BoolValue onlyMe = new BoolValue("OnlyMe", true);
    public static BoolValue onlyOther = new BoolValue("OnlyOther", true);
    public static ListValue mode = new ListValue("Mode", new String[]{"Freddy", "Rabbit", "Amogus"}, "Amogus");
    public static IntegerValue bodyColorR = new IntegerValue("BodyR", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue bodyColorG = new IntegerValue("BodyG", 0,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue bodyColorB = new IntegerValue("BodyB", 0,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue eyeColorR = new IntegerValue("EyeR", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue eyeColorG = new IntegerValue("EyeG", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue eyeColorB = new IntegerValue("EyeB", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue legsColorR = new IntegerValue("LegsR", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue legsColorG = new IntegerValue("LegsG", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue legsColorB = new IntegerValue("LegsB", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
}

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/visual/CustomModel.kt
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.features.module.modules.visual

import net.ccbluex.liquidbounce.features.module.Category
import net.ccbluex.liquidbounce.features.module.Module

object CustomModel : Module("CustomModel", Category.VISUAL) {

    init {
        state = true
    }

    val mode by choices("Mode", arrayOf("Imposter", "Rabbit", "Freddy", "Female", "Invisible"), "Female")
    val rotatePlayer by boolean("RotatePlayer", false)

    val breastPhysics by boolean("BreastPhysics", true) { mode == "Female" }
    val breastGravity by float("BreastGravity", 0.10f, 0.1f..0.5f) { mode == "Female" }
    val breastBounce by float("BreastBounce", 0.6f, 0.1f..1.0f) { mode == "Female" }
    val breastRotation by float("BreastRotation", 0f, 0f..10f) { mode == "Female" }
    val breastNoArmor by boolean("NoRenderInArmor", false) { mode == "Female" }

    override val tag: String
        get() = mode
}